exist-core/src/main/java/org/exist/util/Configuration.java

```diff
@@ -1448,157 +1404,147 @@ private void configureValidation(final Optional<Path> dbHome, final Element vali
-    		
-    		// If the value has not been overriden in a system property, then get it from the configuration
-    		
-    		if( value != null ) {
-                LOG.warn("Configuration value overridden by system property: {}, with value: {}", property, value);
-    		} else {
-    			value = element.getAttribute( attributeName );
-    		}
-    	}
-    	
-    	return( value );
+    private String getConfigAttributeValue(final Element element, final String attributeName) {
+        if (element == null || attributeName == null) {
+            return null;
+        }
+
+        final String property = getAttributeSystemPropertyName(element, attributeName);
+        final String value = System.getProperty(property);
+
+        if (value != null) {
+            LOG.warn("Configuration value overridden by system property: {}, with value: {}", property, value);
+            return value;
+        }
+        // If the value has not been overridden in a system property, then get it from the configuration
+        return element.getAttribute(attributeName);
     }
-    
+
     /**
      * Generates a suitable system property name from the given config attribute and parent element.
-     *
-     * values are of the form org.element.element.....attribute and follow the heirarchical structure of the conf.xml file. 
+     * <p>
+     * values are of the form org.element.element.....attribute and follow the hierarchical structure of the conf.xml file.
      * For example, the db-connection cacheSize property name would be org.exist.db-connection.cacheSize
      *
-     * @param   element        The attribute's parent element
-     * @param   attributeName  The name of the attribute
-     *
-     * @return  The generated system property name
+     * @param element       The attribute's parent element
+     * @param attributeName The name of the attribute
+     * @return The generated system property name
      */
-    private String getAttributeSystemPropertyName( Element element, String attributeName )
-    {
-    	final StringBuilder  	property 	= new StringBuilder( attributeName );
-    	Node			parent		= element.getParentNode();
-    	
-    	property.insert( 0, "." );
-    	property.insert( 0,  element.getLocalName() );
-    	
-    	while( parent != null && parent instanceof Element ) {
-    		final String parentName = ((Element)parent).getLocalName();
-    		
-    		property.insert( 0, "." );
-    		property.insert( 0, parentName );
-    		
-    		parent   = parent.getParentNode();
-    	}
-    	
-    	property.insert( 0, "org." );
-    	
-    	return( property.toString() );
-    }
+    private String getAttributeSystemPropertyName(Element element, String attributeName) {
+        final StringBuilder property = new StringBuilder(attributeName);
+        Node parent = element.getParentNode();
 
+        property.insert(0, ".");
+        property.insert(0, element.getLocalName());
+
+        while (parent instanceof Element) {
```

Comment:
We seem to be missing ` parent != null` check here now?

Reply:
> "At run time, the result of the instanceof operator is true if the value of the RelationalExpression is not null and the reference could be cast to the ReferenceType without raising a ClassCastException. Otherwise the result is false."

Reply:
@dizzzz you are correct